src/slowmokit/methods/metrics/distance_metric/distance_metric.cpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file methods/metrics/distance_metric/distance_metric.hpp
+ *
+ * Easy include to calculate distance metrics
+ */
+
+#include "distance_metric.hpp"
+
+template<class T>
+DistanceMetric<T>::DistanceMetric(std::vector<T> &x, std::vector<T> &y)
+{
+  this->x = x;
+  this->y = y;
+  if (x.size() != y.size())
+  {
+    throw std::domain_error("Size of the two vectors must be same");
+  }
+}
+
+template<class T> double DistanceMetric<T>::euclidean()
+{
+  double distance = 0;
```

Comment:
```suggestion
  long double distance = 0;
```
You will have to downcast `long double` to `double` finally. use [`std::dynamic_cast`](https://en.cppreference.com/w/cpp/language/dynamic_cast)

**UPDATE:** Do this wherever I have added `double -> long double` comments